certora/specs/ValidatorsExitBusOracle.spec

```diff
@@ -0,0 +1,355 @@
+    getResumeSinceTimestamp() returns(uint256) envfree
+    RESUME_ROLE() returns(bytes32) envfree
+    SUBMIT_DATA_ROLE() returns(bytes32) envfree
+    getTotalRequestsProcessed() returns(uint256) envfree
+    getMaxValidatorExitRequestsPerReport() returns(uint256) envfree
+    getLastProcessingRefSlot() returns(uint256) envfree
+    getConsensusVersion() returns(uint256) envfree
+
+    compareBytes(bytes, bytes) returns(bool) envfree
+
+    dataGlobal() returns(uint256, uint256, uint256, uint256, bytes) envfree
+    checkGlobal() returns(uint256) envfree
+
+    getConsensusVersionGlobal() returns(uint256) envfree
+    getRefSlotGlobal() returns(uint256) envfree
+    getRequestsCountGlobal() returns(uint256) envfree
+    getDataFormatGlobal() returns(uint256) envfree
+    getDataGlobal() returns(bytes) envfree
+    matchReportDatas(uint256, uint256, uint256, uint256, bytes) returns(bool) envfree
+    isConsensusMember(address) returns(bool) envfree
+    checkContractVersion(uint256) envfree
+    getRefSlot() returns(uint256) envfree
+    getReportHash() returns(bytes32) envfree
+    hashData() returns(bytes32) envfree
+}
+
+definition excludeInitialize(method f) returns bool =
+    f.selector != initialize(address, address, uint256, uint256).selector;
+
+definition excludeHelper(method f) returns bool =
+    f.selector != submitReportDataHelper(uint256, uint256, uint256, uint256, bytes, uint256).selector;
+
+
+definition matchReportDataWithGlobal(uint256 consensusVersion, uint256 refSlot, uint256 requestsCount, uint256 dataFormat) returns bool = 
+    consensusVersion == getConsensusVersionGlobal()
+    && refSlot == getRefSlotGlobal()
+    && requestsCount == getRequestsCountGlobal()
+    && dataFormat == getDataFormatGlobal()
+    // && compareBytes(dataInput, dataGlobal)
+    ;
+
+
+// STATUS - in progress (can be violated becuase admin can remove it's own role)
+// admin != 0
+invariant adminIsNotZero(address admin)
+    hasRole(DEFAULT_ADMIN_ROLE(), admin) => admin != 0
+
+
+// STATUS - verified
+//  It must not be possible to initialize twice.
+rule cantInitializeTwice(env e, env e2, method f, method g) 
+    filtered { f -> f.selector == initialize(address, address, uint256, uint256).selector,
+               g -> g.selector == initialize(address, address, uint256, uint256).selector } {
+
+    calldataarg args;
+    calldataarg args2;
+
+    f(e, args);
+    g@withrevert(e2, args2);
+
+    assert lastReverted, "Remember, with great power comes great responsibility.";
+}
+
+
+// STATUS - verified
+// Only a user with necessary role can resume within pause timeframe
+rule canResume(env e, env e2) {
+    bool pausedBefore = isPaused(e);
+
+    require e.msg.sender == e2.msg.sender;
+
+    resume@withrevert(e2);
+    bool isReverted = lastReverted;
+
+    bool pausedAfter = isPaused(e2);
+
+    assert (e2.msg.value == 0 
+                && pausedBefore 
+                && e2.block.timestamp < getResumeSinceTimestamp()
```

Comment:
This is equivalent to pausedBefore, no?

Reply:
I checked and it is, i don't remember why it's like this, but it was fixed